src/backend/commands/system/window_hook.cpp

```diff
@@ -0,0 +1,32 @@
+#include "backend/looped_command.hpp"
+#include "memory/byte_patch.hpp"
+#include "pointers.hpp"
+
+namespace big
+{
+	class window_hook : looped_command
```

Comment:
nevermind, why is this a looped_command?

Reply:
Bool command doesn't have an on_enable/on_disable overrides.

![image](https://github.com/YimMenu/YimMenu/assets/24669514/04dca877-007e-4ce1-8460-e5ad42407d0d)

Reply:
https://github.com/YimMenu/YimMenu/blob/ddb12d6e2bc9ec0768550f6760b4459a171c0cd4/src/backend/bool_command.cpp#L13-L38

Please read the impl of bool_command::execute, this will make sense once you do

you need to impl refresh and then check the flag accordingly

Reply:
Well I'll be damned.

Reply:
There are a bunch of commands implemented in this way, I think maybegreat said something along the lines of implementing it like this but my brain my be failing me.

Reply:
that doesnt look like correct usage, i'd argue that on_enable and on_disable should be moved to bool_command,